src/pq_flash_index.cpp

```diff
@@ -649,51 +667,88 @@ inline bool PQFlashIndex<T, LabelT>::point_has_label(uint32_t point_id, LabelT l
 template <typename T, typename LabelT>
 void PQFlashIndex<T, LabelT>::parse_label_file(const std::string &label_file, size_t &num_points_labels)
 {
-    std::ifstream infile(label_file);
+    std::ifstream infile(label_file, std::ios::binary);
     if (infile.fail())
     {
         throw diskann::ANNException(std::string("Failed to open file ") + label_file, -1);
     }
+    infile.seekg(0, std::ios::end);
+    size_t file_size = infile.tellg();
 
-    std::string line, token;
+    std::string buffer(file_size, ' ');
+
+    infile.seekg(0, std::ios::beg);
+    infile.read(&buffer[0], file_size);
```

Comment:
just wonder if others are ok with this changing. For our usage, this file size is not very big, so it is fine.

Reply:
It does seem excessive to read the entire file into memory before counting filters.  Can it be chunked?

Reply:
As the memory usage of the label vector after loading is more than 2/3 of label file size in most scenario (uint16 to represent one label, uint32 for the offset of every node), the implement is just like we need double memory usage in the label loading time, I think it's still fine?